router/core/engine_loader_hooks.go

```diff
@@ -141,10 +141,14 @@ func (f *engineLoaderHooks) OnFinished(ctx context.Context, ds resolve.DataSourc
 			zap.Int("status", responseInfo.StatusCode),
 			zap.Duration("latency", latency),
 		}
-		if responseInfo.Request != nil && responseInfo.Request.URL != nil {
-			fields = append(fields, zap.String("url", responseInfo.Request.URL.String()))
+		path := ds.Name
+		if responseInfo.Request != nil {
```

Comment:
How can Request be nil? I'd expect that the engine always sets it. If not how we do log the request URL? We need to make the request available before we made the request in the engine. Currently, we set everything when request succeed which is not right.

Reply:
Good point! Handled in https://github.com/wundergraph/graphql-go-tools/pull/1016